app/assets/javascripts/discourse/app/components/topic-footer-buttons.gjs

```diff
@@ -43,6 +62,24 @@ export default class TopicFooterButtons extends Component {
   @alias("currentUser.user_option.enable_defer") canDefer;
   @or("topic.archived", "topic.closed", "topic.deleted") inviteDisabled;
 
+  init() {
```

Comment:
I think if you move more stuff into a service you can get rid of all this finagling in this component